File: 0454 - 4Sum II/solution.cpp
class Solution {
public:
    int fourSumCount(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3, vector<int>& nums4) {
        int res = 0;
        unordered_map<int, int> m;
        for(auto i : nums1) for(auto j : nums2) m[i+j]++;
        for(auto i : nums3) for(auto j : nums4) res += m[-i-j];
        return res;
    }
};

File: 0454 - 4Sum II/solution.py
class Solution:
    from collections import Counter
    def fourSumCount(self, nums1: List[int], nums2: List[int], nums3: List[int], nums4: List[int]) -> int:
        c = Counter(i+j for i in nums1 for j in nums2)
        return sum(c[-i-j] for i in nums3 for j in nums4)